spock-spring/src/main/java/org/spockframework/spring/SpringMockTestExecutionListener.java

```diff
@@ -61,6 +64,10 @@ public void beforeTestMethod(TestContext testContext) throws Exception {
     }
   }
 
+  private boolean scanScopedBean(ScanScopedBeans scanScopedBeans, Set<String> scopes, BeanDefinition beanDefinition) {
+    return scanScopedBeans != null && (scopes.size() == 0 || scopes.contains(beanDefinition.getScope()));
```

Comment:
Well the `scanScopedBeans != null` means that scoped scanning is enabled at all. `scopes.size() == 0` means that every scope should be accepted, otherwise we check `scopes.contains(beanDefinition.getScope())` if it matches the scopes to be scanned.

So what should be reduced?